### PR TITLE
fix cursor position after item move command

### LINKED ISSUE #8492: Item movers need some fixes
Hey, I really enjoy item movers (#6823), but I think it needs a little polishing.

## The problems

1. I can't move a function if my cursor is under the argument list. (update: the current feature of "move item up/down" over the argument list is to change their order)
1. My cursor doesn't stay in the same position/column it was before applying the move. It seems that, depending on the function signature (maybe its size?) the cursor goes to a specific column.

These two, in combination, may be annoying, because sometimes my cursor goes to the argument list while I'm moving functions through a file and I have to move the cursor's position back to the function name in order to move it up again.

## Example

```rust
fn main() {
    println!("Hello, world!");
}

fn hi(arg: usize) {}
```

If I move `hi` up, my cursor goes to column 8. If I move it down, it goes to column 2.

## Environment

- Linux
- Rust Stable
- Rust-Analyzer Nightly
- VS Code

### PR DESCRIPTION
**Pull request by eduardocanellas**
Closes #8492

### CODE REVIEW

**Comment by eduardocanellas** [comment]
I started by trying to add tests for the cases I was referring to in #8492. It seems that the `TextEdit` generated by `move_item` is correct for the cases I tested, but the cursor position was not in the right position when I tested on VS Code.

I'll continue looking to see if I can find the cause of the undesired offset.

**Comment by matklad** [comment]
Yeah, unclear why they are not picked up... The VS side of this is here: 

* https://github.com/matklad/rust-analyzer/blob/0d319f296a8ce345a1dc0ec4ca8e9298a52eb605/editors/code/src/snippets.ts#L31-L62
* https://github.com/matklad/rust-analyzer/blob/0d319f296a8ce345a1dc0ec4ca8e9298a52eb605/editors/code/src/commands.ts#L137-L144

**Comment by eduardocanellas** [comment]
Thanks for the links! The problem is certainly on VS side. There's a off-by-one error in something involving `startLine` on `snippets.ts:42`. Maybe with `lineDelta`...

**Comment by eduardocanellas** [comment]
@matklad I think I found the issue. See the following line updating `lineDelta`:

https://github.com/eduardocanellas/rust-analyzer/blob/3bc69f362d8c173ce4748e7e853c0da5183e557a/editors/code/src/snippets.ts#L55

I think that it should increment `lineDelta` instead of overwritting its value. When I changed `=` to `+=` the item movers seemed to work perfectly in the case I was checking. Looks reasonable?

File: crates/ide/src/move_item.rs [hunk 1]
```diff
@@ -465,6 +465,48 @@ fn main() {}
...
+            expect![[r#"
+fn main() {
+    println!("Hello, world!");
+}
+
+fn hi$0(arg: usize) {}
+            "#]],
+            Direction::Down,
+        );
+    }
```
Comment (matklad) [inline crates/ide/src/move_item.rs:508]: Let's remove these tests, if the problem ended up being on the VS Code side. We try to keep the test suite minimal.

Comment (eduardocanellas) [reply crates/ide/src/move_item.rs:508]: Okay, got it.

**Comment by matklad** [comment]
bors r+

Thanks @eduardocanellas

**Comment by lnicola** [comment]
changelog fix (first contribution) fix cursor position after move item command

**Pull request merged**